src/zarr/core/metadata/v2.py

```diff
@@ -108,6 +109,29 @@ def shards(self) -> ChunkCoords | None:
         return None
 
     def to_buffer_dict(self, prototype: BufferPrototype) -> dict[str, Buffer]:
+        def _serialize_fill_value(fv: Any) -> JSON:
```

Comment:
There's several branches of this functions that codecov is reporting as being untested - do tests need adding for these?

Reply:
I'm not really sure what to make of that - the round tripping being done *should* very adequately cover the relevant behaviors. Do you think it would be appropriate to test this function separately as well?

Reply:
It might be that codecov is just incorrect - if you're happy that all these branches are being tested, then I definitely trust you more than codecove 😄